src/box/iproto.cc

```diff
@@ -2138,6 +2138,39 @@ tx_process_rollback(struct cmsg *m)
 	tx_end_msg(msg, &header);
 }
 
+/*
+ * In case the request does not contain a space or identifier but contains a
+ * corresponding name, tries to resolve the name.
+ * A space identifier is mandatory for a CRUD request, so in case the space
+ * identifier is not present in the request and we fail to get an identifier
+ * from the space name (or the space name is missing), we must return with an
+ * error, setting a missing request field diagnostic, just as it was done in
+ * `xrow_decode_dml` before space names in IPROTO requests were introduced.
+ */
+static int
+tx_resolve_space_and_index_name(struct request *dml)
+{
+	if (dml->space_id == 0 && dml->space_name != NULL) {
+		uint32_t space_id = box_space_id_by_name(dml->space_name,
+							 dml->space_name_len);
+		if (space_id != BOX_ID_NIL)
+			dml->space_id = space_id;
+	}
+	if (dml->space_id == 0) {
+		diag_set(ClientError, ER_MISSING_REQUEST_FIELD,
+			 iproto_key_name(IPROTO_SPACE_ID));
+		return -1;
+	}
+	if (dml->index_id == 0 && dml->index_name != NULL) {
```

Comment:
Nit: If the user sets `index_id` to 0, which is a valid id, then it'll be ignored in case `index_name` is present. This isn't critical, but still contradicts the comment to `IPROTO_INDEX_NAME`.

Also, we don't need to resolve the index name for insert and replace requests. Again, not critical, but doesn't look good.

Reply:
> Nit: If the user sets index_id to 0, which is a valid id, then it'll be ignored in case index_name is present. This isn't critical, but still contradicts the comment to IPROTO_INDEX_NAME.

Fixed: now name is preferred over idenitifer. Added test cases for this and updated the design document.

> Also, we don't need to resolve the index name for insert and replace requests. Again, not critical, but doesn't look good.

Fixed, now name is ignored for these requests.